### PR TITLE
3dstreet editor first ver

### PR DESCRIPTION
**Pull request by Algorush**

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,4 +1,4 @@
 {
-  "plugins": ["transform-class-properties"],
+  "plugins": ["transform-class-properties", "transform-object-rest-spread"],
```
Comment (srothst1) [inline .babelrc:2]: I'm not sure exactly what this line does. Could you give me a high level overview?

File: package.json [hunk 2]
```diff
@@ -79,6 +79,8 @@
     "wysiwyg"
   ],
   "devDependencies": {
+    "@babel/preset-stage-3": "^7.8.3",
+    "babel-plugin-transform-object-rest-spread": "^6.26.0",
```
Comment (srothst1) [inline package.json:83]: Why are these plugins necessary?

File: src/components/scenegraph/Toolbar.js [hunk 3]
```diff
@@ -44,6 +44,10 @@ export default class Toolbar extends React.Component {
       isPlaying: false
     };
   }
+  makeScreenshot() {
```
Comment (srothst1) [inline src/components/scenegraph/Toolbar.js:47]: Looks good 👍 I am glad that we are going with perspective here.

File: src/components/scenegraph/Toolbar.js [hunk 4]
```diff
@@ -113,12 +117,18 @@ export default class Toolbar extends React.Component {
...
+            onClick={this.makeScreenshot}>
+            <div></div>
+          </a>
           <a
             className="gltfIcon"
             title="Export to GLTF"
             onClick={this.exportSceneToGLTF}>
-            <img src={process.env.NODE_ENV === 'production' ? 'https://aframe.io/aframe-inspector/assets/gltf.svg' : '../assets/gltf.svg'} />
-          </a>
+            <div></div>
```
Comment (srothst1) [inline src/components/scenegraph/Toolbar.js:132]: Is this an empty `div`?

File: src/components/viewport/CameraToolbar.js [hunk 5]
```diff
@@ -45,9 +45,19 @@ export default class CameraToolbar extends React.Component {
   }
 
   render() {
+    const customStyles = {
+      dropdownIndicator: (provided, state) => ({
+        backgroundImage: state.isSelected ? 'url("/assets/arrow_up.svg")': 'url("/assets/arrow_down.svg")'
+      }),
+      menuList: (provided, state) => ({
```
Comment (srothst1) [inline src/components/viewport/CameraToolbar.js:52]: This looks great! Some next steps here might be changing the options/wording in the camera selector. However, we can leave that change for a later PR.

**Review by srothst1 — COMMENTED** [review]
Really solid work @Algorush! I am particularly impressed by your attention to detail. My review contains a few questions and quick comments. @kfarr please feel free to add your input here as well.

**Comment by srothst1** [comment]
@kfarr this morning we discussed some methods to streamline access to the aframe inspector. It seems like that issue has already been resolved - when I run this locally, I immediately see the re-skinned aframe inspector.

File: .babelrc (hunk 1 above)
Comment (Algorush) [reply .babelrc:2]: Hi! aframe-inspector uses the react-select library (https://react-select.com/home) for the dropdown menu with camera modes. To be able to change the styles of some objects in the dropdown menu, the react-select documentation suggests using the following code:
`const customStyles = {
  option: (provided, state) => ({
    ...provided,
    borderBottom: '1px dotted pink',
    color: state.isSelected ? 'red' : 'blue',
    padding: 20,
  }),...`
As you can see, this uses the rest operator, which is not supported in the version of Babel that comes with aframe-inspector. I saw advice on StackOverflow that the best way to fix this is to use this simple plugin that adds rest support.

File: src/components/viewport/CameraToolbar.js (hunk 5 above)
Comment (Algorush) [reply src/components/viewport/CameraToolbar.js:52]: Looks good, but for some reason, this code doesn't work yet :), I haven't figured out why yet

File: src/components/scenegraph/Toolbar.js (hunk 4 above)
Comment (Algorush) [reply src/components/scenegraph/Toolbar.js:132]: Yes, the background picture for the button is located inside the inner div. Maybe it was possible to do it for the a element, but I decided this way :)

**Comment by kfarr** [comment]
I did a netlify deploy so we can share this with Daria https://scintillating-donut-29d0e2.netlify.app/examples/ . Final feedback from her to merge this.

**Comment by kfarr** [comment]
Design review feedback:
https://docs.google.com/document/d/1nIx43M2ZLGtELJNmCHqlmjTDDRIZ10ohQsDP82t5QzU/edit

File: .babelrc (hunk 1 above)
Comment (srothst1) [reply .babelrc:2]: This make sense! Thank you for the clarification 😀

File: src/components/viewport/CameraToolbar.js (hunk 5 above)
Comment (srothst1) [reply src/components/viewport/CameraToolbar.js:52]: Got it 👍

**Comment by srothst1** [comment]
Thanks @Algorush for all of your hard work! This PR is good to go 👍

**Pull request merged**